app/controllers/api/journeys_controller.rb

```diff
@@ -136,22 +136,24 @@ def update_journey_attributes
     end
 
     def create_event
-      # Logs the event for posterity and the immutable event log
-      event = journey.events.create!(
-        event_name: action_name,
-        client_timestamp: Time.zone.parse(data_params.dig(:attributes, :timestamp)),
-        details: {
-          data_params: data_params,
-          supplier_id: supplier.id,
-        },
-      )
-      create_generic_event(event)
-    end
-
-    def create_generic_event(event)
-      generic_event = GenericEvent.from_event(event)
-      generic_event.save!
-      event.update!(generic_event_id: generic_event.id)
+      now = Time.zone.now.iso8601
+
+      event_attributes = {
+        eventable: journey,
+        occurred_at: now,
+        recorded_at: now,
+        notes: 'Automatically generated event',
+        details: {},
+        supplier_id: supplier.id,
```

Comment:
This `supplier` getter is different from the one we have for the MoveProposed and MoveRequested events. It gives preference to the id as an attribute coming from the json:api relationship and only falls back on the supplier coming from the application owner.

This override could be confusing the meaning of this id (indicating who owns instead of who created the event)

Reply:
I'm being cheeky getting this related/unrelated change in @martyn-w . If it's too clutterful for review I'll remove it :).